### PR TITLE
Dont cache bust urls

### PR DESCRIPTION
**Pull request by gauntface**
R: @jeffposnick @addyosmani 

Adds dontCacheBustURLsMatching parameter

### CODE REVIEW

File: packages/sw-build/src/lib/generate-sw.js [hunk 1]
```diff
@@ -55,6 +55,9 @@ const errors = require('./errors');
  * @param {number} [input.maximumFileSizeToCacheInBytes] An optional number to
  * define the maximum file size to consider whether the file should be
  * precached. (Defaults to 2MB).
+ * @param {Regex} [input.dontCacheBustUrlsMatching] An optional regex that will
```
Comment (jeffposnick) [inline packages/sw-build/src/lib/generate-sw.js]: Regex => RegExp

File: packages/sw-build/src/lib/get-file-manifest-entries.js [hunk 2]
```diff
@@ -30,6 +30,9 @@ const getStringDetails = require('./utils/get-string-details');
  * @param {number} [input.maximumFileSizeToCacheInBytes] An optional number to
  * define the maximum file size to consider whether the file should be
  * precached. (Defaults to 2MB).
+ * @param {Regex} [input.dontCacheBustUrlsMatching] An optional regex that will
```
Comment (jeffposnick) [inline packages/sw-build/src/lib/get-file-manifest-entries.js]: Regex => RegExp

File: packages/sw-build/src/lib/generate-sw.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ const errors = require('./errors');
  * @param {number} [input.maximumFileSizeToCacheInBytes] An optional number to
  * define the maximum file size to consider whether the file should be
  * precached. (Defaults to 2MB).
- * @param {Regex} [input.dontCacheBustUrlsMatching] An optional regex that will
+ * @param {Regexp} [input.dontCacheBustUrlsMatching] An optional regex that will
```
Comment (jeffposnick) [inline packages/sw-build/src/lib/generate-sw.js]: I don't know how particular JSDoc is, or whether it even matters in terms of how it generates the docs, but the capitalization of the JavaScript class is [`RegExp`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp). Feel free to ignore if it doesn't matter.

Comment (addyosmani) [inline packages/sw-build/src/lib/generate-sw.js]: I assumed it was RegExp but similarly don't know how well JSDoc would trip up on casing here. Deferring to Matt.

**Pull request merged**